src/pint/models/pulsar_binary.py

```diff
@@ -185,7 +187,31 @@ def apply_units(self):
             bparObj.value = bparObj.value * u.Unit(bparObj.units)
 
     def update_binary_object(self, toas=None, acc_delay=None):
-        """Update binary object instance for this set of parameters/toas."""
+        """Update stand alone binary's parameters and toas from PINT-facing object.
+
+        This function passes the PINT-facing object's parameter values and TOAs
+        to the stand-alone binary object. If the TOAs are not provided, it only
+        updates the parameters not the TOAs.
+
+        Parameters
+        ----------
+        toas : `toa.TOAs` object, optional
+            The TOAs that need to pass to the stand alone model.Default value is
+            None. If toas is None, this function only updates the parameter value.
+            If 'acc_delay' is not provided, the stand alone binary receives the
+            standard barycentered TOAs.
+
+        acc_delay: `Numpy.ndarray`
+            If provided, TOAs will be corrected by provided acc_delay instead of
+            the standard barycentering. The stand alone binary receives the
+            input TOAs - acc_delay.
+
+        Note
+        ----
+        If the 'toas' are not provided, the stand alone binary model will use
+        the TOAs were passed to it from last interation or no TOAs for stand
+        alone binary model at all.   
```

Comment:
I'm confused by this - can this function accept arbitrary TOAs object, or is there some requirement that they already be barycentered? When no TOAs are passed, are the same TOAs as before used (so that changing the sky position would result in different residuals) or is some kind of intermediate result stored?

Reply:
If the acc_delay is not given:
     This function does the barycentering itself. It accepts the observatory TOAs object.
If the acc_delay is given:
    This function pass the observatory - acc_delay to the stand alone binary model. 

The standalone model save the "barycentered TOA" in itself. If one changes the sky position, without passing TOAs. the binary delay would not change.

Reply:
I know this could be risky. That is why I want to remove the stand-alone binary model.